source/Applications/Advanced/TransformViaArucoMarker/TransformViaArucoMarker.cpp

```diff
@@ -0,0 +1,308 @@
+
+        int w = pointCloud.width();
+        int h = pointCloud.height();
+        const auto points = pointCloud.copyPointsXYZ();
+
+        std::vector<cv::Point3d> markerPoints3D;
+        markerPoints3D.reserve(markerPoints2D.size());
+        for(const auto &point2D : markerPoints2D)
+        {
+            float x = point2D.x;
+            float y = point2D.y;
+
+            int x1 = clamp(static_cast<int>(std::floor(x)), 0, w - 1);
+            int x2 = clamp(static_cast<int>(std::ceil(x)), 0, w - 1);
+            int y1 = clamp(static_cast<int>(std::floor(y)), 0, h - 1);
+            int y2 = clamp(static_cast<int>(std::ceil(y)), 0, h - 1);
+
+            auto q11 = points(y1, x1);
+            auto q12 = points(y2, x1);
+            auto q21 = points(y1, x2);
+            auto q22 = points(y2, x2);
+
+            float a = (x2 - x) / (x2 - x1); // NOLINT(cppcoreguidelines-narrowing-conversions)
+            float b = (x - x1) / (x2 - x1); // NOLINT(cppcoreguidelines-narrowing-conversions)
+
+            const cv::Point3f fxy1{ a * q11.x + b * q21.x, a * q11.y + b * q21.y, a * q11.z + b * q21.z };
+            const cv::Point3f fxy2{ a * q12.x + b * q22.x, a * q12.y + b * q22.y, a * q12.z + b * q22.z };
+
+            float c = (y2 - y) / (y2 - y1); // NOLINT(cppcoreguidelines-narrowing-conversions)
+            float d = (y - y1) / (y2 - y1); // NOLINT(cppcoreguidelines-narrowing-conversions)
+            markerPoints3D.emplace_back(
+                cv::Point3f{ c * fxy1.x + d * fxy2.x, c * fxy1.y + d * fxy2.y, c * fxy1.z + d * fxy2.z });
+        }
+
+        return markerPoints3D;
+    }
+
+    Zivid::Matrix4x4 transformationMatrix(const cv::Matx33d &rotationMatrix, const cv::Vec3d &translationVector)
+    {
+        auto transformMatrix = Zivid::Matrix4x4{ 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1 };
+
+        for(size_t row = 0; row < rotationMatrix.rows; row++)
+        {
+            for(size_t col = 0; col < rotationMatrix.cols; col++)
+            {
+                transformMatrix(row, col) = rotationMatrix(row, col);
+            }
+        }
+
+        transformMatrix(0, 3) = translationVector[0];
+        transformMatrix(1, 3) = translationVector[1];
+        transformMatrix(2, 3) = translationVector[2];
+
+        return transformMatrix;
+    }
+
+    void displayMarkerPose(const Zivid::PointCloud &pointCloud,
+                           const std::vector<cv::Point2f> &corners2D,
+                           const std::vector<cv::Point2f> &center2D)
+    {
+        auto rgbaImage = cv::Mat(pointCloud.height(), pointCloud.width(), CV_8UC4); // NOLINT(hicpp-signed-bitwise)
+        auto *ptr =
+            reinterpret_cast<Zivid::ColorRGBA *>(rgbaImage.data); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
+        pointCloud.copyData(ptr);
+        auto bgr = cv::Mat(pointCloud.height(), pointCloud.width(), CV_8UC4); // NOLINT(hicpp-signed-bitwise)
+        cv::cvtColor(rgbaImage, bgr, cv::COLOR_BGR2RGB);
```

Comment:
You have a function pointCloudToBGR(), why not use it?

Reply:
good point. Guess I mixed up what functions I had available then I wrote the displayMarkerPose. Using the pointCloudToBGR() now.